src/Microsoft.AspNet.Mvc.Core/OptionDescriptors/OptionDescriptorExtensions.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Linq;
+using System.Collections.Generic;
+using Microsoft.AspNet.Mvc.OptionDescriptors;
+
+namespace Microsoft.AspNet.Mvc
+{
+    /// <summary>
+    /// Extension methods for collections of option descriptors.
+    /// </summary>
+    public static class OptionDescriptorExtensions
+    {
+        /// <summary>
+        /// Returns the only instance of <typeparamref name="TInstance"/> from a sequence of option descriptors.
+        /// </summary>
+        /// <typeparam name="TInstance">The type of the instance to find.</typeparam>
+        /// <param name="descriptors">A sequence of <see cref="IOptionDescriptor{object}"/>.</param>
+        /// <returns>The only instance of <typeparamref name="TInstance"/> from a sequence of 
+        /// <see cref="IOptionDescriptor{object}"/>.</returns>
+        /// <exception cref="InvalidOperationException"> 
+        /// Thrown if there is not exactly one <typeparamref name="TInstance"/> in the sequence.</exception>
+        public static TInstance InstanceOf<TInstance>([NotNull] this IEnumerable<IOptionDescriptor<object>> descriptors)
+        {
+            var instance = descriptors.Single(d => d.OptionType == typeof(TInstance)
+                && d.Instance != null).Instance;
```

Comment:
I don't think there's a hard and fast rule about using single letter variable names for lambda parameters.  I notice it's about half and half in Microsoft.AspNet.Mvc.Core. Inclined to leave it as is for now, unless @yishaigalatzer thinks otherwise.